prov/gni/test/freelist.c

```diff
@@ -124,6 +124,37 @@ Test(gnix_freelist, freelist_refill_test)
 	_gnix_fl_destroy(&fl);
 }
 
+Test(gnix_freelist, freelist_zero_refill_test)
+{
+	struct gnix_freelist fl;
+	int i, ret;
+	const int num_elems = 71;
+	struct dlist_entry *elems[num_elems + 1];
+	const int refill_size = 0;
+
+	/* non-optimized code may not zero structures */
+	memset(&fl, 0x0, sizeof(struct gnix_freelist));
```

Comment:
Oh well now I feel silly... So I remember this issue vaguely. In default debug builds, there was no guarantee that the structure would be zeroed out. Stack memory was left in prior state, so it would fail in debug dev builds. 

Nevermind this comment. It can't be fixed anywhere but here because the freelists are allocated on stack.